test-framework/junit5/src/main/java/io/quarkus/test/junit/internal/CustomListConverter.java

```diff
@@ -0,0 +1,32 @@
+package io.quarkus.test.junit.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.thoughtworks.xstream.converters.collections.CollectionConverter;
+import com.thoughtworks.xstream.mapper.Mapper;
+
+/**
+ * A custom List converter that always uses ArrayList for unmarshalling.
+ * This is probably not semantically correct 100% of the time, but it's likely fine
+ * for all the cases where we are using marshalling / unmarshalling.
+ *
+ * The reason for doing this is to avoid XStream causing illegal access issues
+ * for internal JDK lists
+ */
+public class CustomListConverter extends CollectionConverter {
+
+    public CustomListConverter(Mapper mapper) {
+        super(mapper);
+    }
+
+    @Override
+    public boolean canConvert(Class type) {
+        return List.class.isAssignableFrom(type);
```

Comment:
We might need to be more strict here

Reply:
Yeah, if doable, I think we'd better list the problematic ones rather than be too broad.
That won't be perfect but less dangerous.